app/models/manageiq/providers/ansible_tower/automation_manager/cloud_credential.rb

```diff
@@ -1,2 +1,8 @@
 class ManageIQ::Providers::AnsibleTower::AutomationManager::CloudCredential < ManageIQ::Providers::AnsibleTower::AutomationManager::Credential
+  def self.credential_types
+    subclasses.sort_by(&:name).each_with_object({}) do |subclass, credential_types|
+      provider_name = subclass.name.demodulize.split('Credential').first
```

Comment:
We have the same concept on other models already...
https://github.com/ManageIQ/manageiq/blob/master/app/models/ext_management_system.rb#L20-L26
https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/redhat/infra_manager.rb#L32-L38